framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -918,40 +911,22 @@ private boolean hasDependentType(AnnotatedTypeMirror atm) {
         if (atm == null) {
             return false;
         }
-        Boolean b = new ContainsDependentType().visit(atm);
-        if (b == null) {
-            return false;
-        }
+        boolean b =
+                new SimpleAnnotatedTypeScanner<>(
+                                (type, p) -> {
+                                    for (AnnotationMirror am : type.getAnnotations()) {
```

Comment:
Could this be rewritten something like `type.getAnnotations().stream().anyMatch(am -> isExpressionAnno(am))`?